File: web/types/cliente.ts
export interface Cliente {
  id: string; // Agora ele é obrigatório na leitura (GET)
  nome: string;
  cpf: string;
  telefone: string;
  email: string;
  endereco: string;
}

export type ClienteCreate = Omit<Cliente, 'id'>;

File: backend/src/main/java/br/com/guerreiro_dev/backend/domain/Cliente.java
package br.com.guerreiro_dev.backend.domain;

import jakarta.persistence.*;
import lombok.*;

import java.io.Serial;
import java.io.Serializable;
import java.util.UUID;

@Getter
@Setter
@EqualsAndHashCode(onlyExplicitlyIncluded = true)
@Entity
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "tb_cliente")
public class Cliente implements Serializable {

    @Serial
    private static final long serialVersionUID = 1L;

    @EqualsAndHashCode.Include
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;
    private String nome;
    private String cpf;
    private String telefone;
    private String email;
    private String endereco;



}


